### PR TITLE
Handle has-many in OnlineQuery request and response

### PR DESCRIPTION
**Pull request by rooftoofwoof**
- [x] allows has-manys to be passed into OnlineQueryParams as a list of structs 
- [x] OnlineQuery unmarshals a has-many result into a list of structs
- [x] Tests with all data types for both serialization and deserialization

Out of scope
- [ ] Has-manys in OnlineQueryBulk

### CODE REVIEW

File: internal/codegen_utils.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"strings"
+)
+
+type hasManyJoin struct {
+	localKey   string
+	foreignKey string
+}
+
+func isHasMany(field reflect.StructField) bool {
+	// TODO: Really the tag is used only for UnmarshalInto.
```
Comment (rooftoofwoof) [inline internal/codegen_utils.go:18]: remove comment

Comment (rooftoofwoof) [reply internal/codegen_utils.go:18]: let's delete this file since these methods were not ultimately used

File: internal/codegen_utils.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+func getHasManyJoin(field reflect.StructField) hasManyJoin {
+	tags := strings.Split(field.Tag.Get("has_many"), ",")
+	return hasManyJoin{
+		localKey:   tags[0],
+		foreignKey: tags[1],
+	}
+}
+
+func ResolveFeatureName(field reflect.StructField) (string, error) {
```
Comment (rooftoofwoof) [inline internal/codegen_utils.go:34]: let's move this back to where it came from

File: unmarshal.go [hunk 3]
```diff
@@ -57,10 +57,73 @@ func setFeatureSingle(field reflect.Value, fqn string, value any) error {
...
+	numRows := len(values[0])
+
+	newValues := make([]map[string]any, numRows)
+	for rowIdx := 0; rowIdx < numRows; rowIdx++ {
+		newRow := make(map[string]any)
+		for colIdx, colName := range columns {
+			cell := values[colIdx][rowIdx]
+			colParts := strings.Split(colName, ".")
+			fieldName := colParts[len(colParts)-1]
+			newRow[fieldName] = cell
```
Comment (rooftoofwoof) [inline unmarshal.go:112]: inline cell

File: internal/tests/integration/query_test.go [hunk 4]
```diff
@@ -71,3 +72,35 @@ func TestOnlineQueryAllTypesUnmarshalling(t *testing.T) {
...
+	}
+	params := chalk.OnlineQueryParams{}.
+		WithInput(testFeatures.Series.Id, "seed").
+		WithInput(testFeatures.Series.Investors, investorsInput).
+		WithOutputs(testFeatures.Series.Name, testFeatures.Series.Investors)
+
+	var resultSeries series
+	res, err := client.OnlineQuery(params, &resultSeries)
+	assert.NoError(t, err)
+	assert.Equal(t, len(investorsInput), len(*resultSeries.Investors))
```
Comment (rooftoofwoof) [inline internal/tests/integration/query_test.go:97]: check each individual resultSeries.Investors

File: internal/unmarshal.go [hunk 5]
```diff
@@ -242,27 +245,69 @@ func GetReflectValue(value any, typ reflect.Type) (*reflect.Value, error) {
...
+						)
+					}
+					for _, tag := range intTags {
+						bucketFqn := fmt.Sprintf("%s__%d__", resolved, tag)
+						nameToField[bucketFqn] = structValue.Field(i)
+					}
+				}
 			}
 			for k, v := range mapz {
+				// FIXME: Convert k to base windowed feature FQN
```
Comment (rooftoofwoof) [inline internal/unmarshal.go:267]: this is done. remove

File: serializers.go [hunk 6]
```diff
@@ -2,16 +2,60 @@ package chalk
...
 	"github.com/cockroachdb/errors"
 	"github.com/samber/lo"
 	"google.golang.org/protobuf/types/known/structpb"
 	"google.golang.org/protobuf/types/known/timestamppb"
+	"reflect"
 	"time"
 )
 
-func (p OnlineQueryParams) serialize() internal.OnlineQueryRequestSerialized {
+func getConvertedValues(values any) (any, error) {
```
Comment (rooftoofwoof) [inline serializers.go:16]: rename to convertIfHasMany

File: unmarshal.go [hunk 7]
```diff
@@ -22,45 +21,85 @@ func setFeatureSingle(field reflect.Value, fqn string, value any) error {
...
+			return errors.Wrapf(err, "error setting map entry value for feature '%s'", fqn)
 		}
-		field.SetMapIndex(tagValue, internal.ReflectPtr(*rVal))
 		return nil
 	} else {
 		return fmt.Errorf("expected a pointer type for feature '%s', found %s", fqn, field.Type().Kind())
 	}
 }
 
+func getConvertedValue(value any) (any, error) {
```
Comment (rooftoofwoof) [inline unmarshal.go:37]: rename convertIfHasMany

File: internal/unmarshal.go [hunk 8]
```diff
@@ -60,6 +60,30 @@ func IsDataclass(field reflect.Value) bool {
...
+		// Not a dataclass nor a has-many feature.
+		return false
+	}
+
+	if typ == reflect.TypeOf(time.Time{}) {
+		return false
+	}
+
+	if IsTypeDataclass(typ) {
+		// Don't manually serialize dataclasses. Dataclasses need to be serialized
```
Comment (rooftoofwoof) [inline internal/unmarshal.go:78]: This comment should be moved out but at the same time check whether the logic still stands where it should be moved.

**Review by AndyMoreland — APPROVED** [review]

**Pull request merged**